client/me/purchases/remove-plan-dialog/index.tsx

```diff
@@ -0,0 +1,205 @@
+import { Dialog, Gridicon } from '@automattic/components';
+import { createElement, createInterpolateElement } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { SiteScreenshot } from '../site-screenshot/';
+import getPlanFeatures from './get-plan-features';
+import './style.scss';
+
+interface RemovePlanDialogProps {
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	site: SiteExcerptData;
+	hasDomain: boolean;
+	isRefundable: boolean;
+	primaryDomain: string;
+	wpcomURL: string;
+}
+
+export const RemovePlanDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+	hasDomain,
+	isRefundable,
+	primaryDomain,
+	wpcomURL,
+}: RemovePlanDialogProps ) => {
+	const translate = useTranslate();
+	const siteName = site.name ?? '';
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const isComingSoon = site.is_coming_soon;
+	const isPrivate = site.is_private;
+	const launchedStatus = site.launch_status === 'launched' ? true : false;
+	const shouldUseSiteThumbnail =
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	const subTitle = isRefundable
+		? translate( 'If you cancel your plan, once it expires, you will lose:' )
+		: translate( 'If you cancel your plan, you will lose:' );
```

Comment:
These are still backwards.  But otherwise, this part is now working correctly.

Reply:
Actually, not entirely sure `isRefundable` is the correct thing to check on its own -- if the subscription isn't refundable but auto-renew is off, this dialog will still result in the plan being removed immediately, I think (not when it expires)?  I haven't traced this code though.

Reply:
could you please elaborate this? `isRefundable` is a bool prop now in the `RemovePlanDialog` componente. The value is calculated by using the function `isRefundable()` in the parent components.

Reply:
I think what I'm saying is that the following is the text/behavior I would expect:
- **If the subscription is refundable:**  Text should say "If you cancel your plan, you will lose" and the end result is that the subscription should be removed immediately
- **If the subscription is not refundable and auto-renew is on:**  Text should say "If you cancel your plan, once it expires, you will lose" and the end result is that the subscription should have auto-renew turned off
- **If the subscription is not refundable and auto-renew is off:**  Text should say "If you cancel your plan, you will lose" and the end result is that the subscription should be removed immediately

At least I'm pretty sure that's what it should be.

Reply:
this logic has been implemented along with the requested copy